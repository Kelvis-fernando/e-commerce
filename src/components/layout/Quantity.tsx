import useQuantity from "../../hooks/layout/useQuantity";
import { QuantityContainer } from "../../styles/layout/quantity";

const Quantity = () => {
  const { handleIncrement, handleDecrement, quantity } = useQuantity();

  return (
    <QuantityContainer>
      <button onClick={handleIncrement}>+</button>
      <div>{quantity}</div>
      <button onClick={handleDecrement}>-</button>
    </QuantityContainer>
  );
};

export default Quantity;
