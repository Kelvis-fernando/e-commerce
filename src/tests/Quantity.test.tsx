import React from "react";
import { render } from "@testing-library/react";
import Quantity from "../components/layout/Quantity";

describe("Quantity", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<Quantity />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
