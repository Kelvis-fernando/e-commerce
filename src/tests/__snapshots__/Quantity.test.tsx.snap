// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quantity should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hHqvzg"
  >
    <button>
      +
    </button>
    <div>
      quantity
    </div>
    <button>
      -
    </button>
  </div>
</div>
`;
